### ISSUE TITLE
TimeZone conflicts while deserializing

### ISSUE THREAD

**Issue by Shriyanshagro** [plone/plone.restapi#377]
there should be check in data type if two value being compared are coming from different functions
[codeline this bug refers to](https://github.com/plone/plone.restapi/blob/master/src/plone/restapi/deserializer/dxcontent.py#L81)

Timezone to DateTime value only be assigned dynamically if tzinfo not present in value.
[codeline this issue refer to](https://github.com/plone/plone.restapi/blame/master/src/plone/restapi/deserializer/dxfields.py#L49)

Also, tzinfo must not be always set to localZone() but to UTC, as sometimes server machine DateTime is not correctly set.

[Error it cause in plone.importexport](https://github.com/Shriyanshagro/plone.importexport/issues/4)

**Comment by tisto** [comment]
@Shriyanshagro please provide us with a more specific and detailed error report. We need a curl example call (feel free to use plonedemo.kitconcept.com) and the current response and the expected response.

**Comment by Shriyanshagro** [comment]
I think curl call is not needed here.

TypeError: "can't compare offset-naive and offset-aware datetimes"
[Codeline](https://github.com/plone/plone.restapi/blob/master/src/plone/restapi/deserializer/dxcontent.py#L81) in reastapi responsible for this error
Example values:
var dm.get() >> datetime.datetime(2017, 6, 28, 17, 0, tzinfo=<DstTzInfo 'Asia/Kolkata' IST+5:30:00 STD>)
var value >> datetime.datetime(2017, 6, 28, 17, 0)


While tracing back the error:
[codeline](https://github.com/plone/plone.restapi/blame/master/src/plone/restapi/deserializer/dxfields.py#L49) which causes the bug
_Suggested Solution:_ Timezone to DateTime value only be assigned dynamically if tzinfo not present in value at the first place. 
Also, my suggestion is tzinfo must not be always set to localZone() but to UTC, as sometimes server machine DateTime is not correctly set.

**Comment by lukasgraf** [comment]
> I think curl call is not needed here.

Yes it is. At this point, it's totally unclear to me what the bug in `plone.restapi` is supposed to be.

Please provide a clear description of:
1) What you did (setup in terms of existing content, and the HTTP call to the API that produced the problem)
2) What result you were expecting
3) The result you got instead

Ideally this would be in the form of a **failing testcase** for `plone.restapi`, but a precise narrative description works too.

But we do need, as an absolute minimum, **performed actions**, **expected results** and **actual results** in order to understand the erroneous behavior you're seeing and confirm this as a bug, and to even start discussing the issue. This needs to happen before we talk suggested solutions.

**Comment by Shriyanshagro** [comment]
Okay, the potential problem here is with the adapter provided by plone.restapi.interfaces

When I do a curl-
curl -i -X PATCH http://localhost:8080/P2/events/new-event -H "Accept: application/json" -H "Content-Type: application/jon" --data-raw '{"start": "2017-08-20T19:00:00+05:30"}' --user admin:admin
It works as expected.

But when we use adapter `deserializer = queryMultiAdapter((context, request), IDeserializeFromJson)`
 to deserialize json data `data = {u'end': u'2017-10-13T20:00:00+05:30', u'start': u'2017-08-13T19:00:00+05:30', u'@type': u'Event', u'path': u'P2/events/deadline-for-talk-submission',}
`
It throws me this error `*** BadRequest: [{'field': 'start', 'message': "can't compare offset-naive and offset-aware datetimes", 'error': TypeError("can't compare offset-naive and offset-aware datetimes",)}, {'field': 'end', 'message': "can't compare offset-naive and offset-aware datetimes", 'error': TypeError("can't compare offset-naive and offset-aware datetimes",)}]`

It is due to the fact that - "Zope DateTime handles all timezone information as offsets from GMT
So while converting to python datetime object, it losses some info like timezone information''

**Comment by Shriyanshagro** [comment]
@lukasgraf  and @tisto. Is my previous example good enough to understand or you need more explanation on it?

**Comment by Shriyanshagro** [comment]
@lukasgraf , @tisto ,  @frapell  and @djay  . I have pushed a Test in the previous commit to describe the underlying issue here. Complete [log](https://pastebin.com/Rwyn2A5J) of failing testcase.

**Comment by djay** [comment]
@Shriyanshagro as mentioned earlier. Re-edit your first comment to make it blindingly obvious what the issue is. I still have no clue. If you don't make it easier for others to understand then it is less likely to be fixed. You have been given excellent advice on how to do this by @lukasgraf. Just ask yourself, if you looked at that test case, would that make it obvious what the problem is?

**Comment by Shriyanshagro** [comment]
@lukasgraf , @tisto  , @frapell  and @djay  

Seems like people can't follow what I'm trying to establish here. 
So I opened a new issue with sufficient description, cases, possible errors and cause.

Please refer to this new opened [issue](https://github.com/plone/plone.restapi/issues/396)
